great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -1031,8 +1031,8 @@ def _build_selectable_from_batch_spec(self, batch_spec):
         return sa.select("*").select_from(sa.text(table_name)).where(split_clause)
 
     def get_batch_data_and_markers(
-        self, batch_spec
-    ) -> Tuple[SqlAlchemyBatchData, BatchMarkers]:
+        self, batch_spec: BatchSpec
+    ) -> Tuple[Any, BatchMarkers]:
```

Comment:
What is happening here?

Reply:
The previous interface was wrong -- not properly inclusive of all possibilities.

Reply:
I believe that this was correct before; if there is a reason to loosen the interface (which should be only at the parent if it's done), it should be to `BatchData` vs. `Any`

Reply:
@jcampbell Unfortunately, `SqlAlchemyBatchData` does not extend `BatchData` -- so picked `Any` to remedy it.

Reply:
Thank you for clarifying. We can add this to a 'future clean up' section.